src/calendar-list/item.tsx

```diff
@@ -21,120 +20,100 @@ export type CalendarListItemProps = CalendarProps & {
   scrollToMonth?: (date: XDate) => void;
 };
 
-type CalendarListItemState = {
-  hideArrows: boolean;
-  hideExtraDays: boolean;
-};
-
-class CalendarListItem extends Component<CalendarListItemProps, CalendarListItemState> {
-  static displayName = 'CalendarListItem';
-
-  static propTypes = {
-    ...Calendar.propTypes,
-    item: PropTypes.any,
-    calendarWidth: PropTypes.number,
-    calendarHeight: PropTypes.number,
-    horizontal: PropTypes.bool
-  };
-
-  static defaultProps = {
-    hideArrows: true,
-    hideExtraDays: true
-  };
-
-  style: any;
-
-  constructor(props: CalendarListItemProps) {
-    super(props);
-
-    this.style = styleConstructor(props.theme);
-  }
-
-  shouldComponentUpdate(nextProps: CalendarListItemProps) {
-    const r1 = this.props.item;
-    const r2 = nextProps.item;
-
-    return !sameMonth(r1, r2) || !!(r2.propBump && r2.propBump !== r1.propBump);
-  }
-
-  onPressArrowLeft = (method: () => void, month: XDate) => {
-    const {onPressArrowLeft, scrollToMonth} = this.props;
-    const monthClone = month.clone();
-
-    if (onPressArrowLeft) {
-      onPressArrowLeft(method, monthClone);
-    } else if (scrollToMonth) {
-      const currentMonth = monthClone.getMonth();
-      monthClone.addMonths(-1);
-
-      // Make sure we actually get the previous month, not just 30 days before currentMonth.
-      while (monthClone.getMonth() === currentMonth) {
-        monthClone.setDate(monthClone.getDate() - 1);
+const CalendarListItem = (props: CalendarListItemProps) => {  
```

Comment:
It's very possible you still need to wrap this with React.memo so the component behave as a pure component